### PR TITLE
support removing and falling back to a previous template

### PR DESCRIPTION
**Pull request by JoviDeCroock**

### CODE REVIEW

File: src/dispatcher/index.ts [hunk 1]
```diff
@@ -100,35 +105,33 @@ const createDispatcher = () => {
...
+        metaQueue.splice(currentMetaIndex++, 0, payload as MetaPayload);
+      } else {
+        linkQueue.push(payload);
       }
     },
     _removeFromQueue: (type: HeadType, payload: MetaPayload | string) => {
-      if (type === 'title') {
-        titleQueue.splice(titleQueue.indexOf(payload as string), 1);
-        if (!isServerSide)
+      if (type === TITLE || type === TEMPLATE) {
```
Comment (JoviDeCroock) [inline src/dispatcher/index.ts:129]: CC @gustavpursche added some cases here to fallback when we remove title or template on index 0

File: src/dispatcher/index.ts [hunk 2]
```diff
@@ -163,14 +166,15 @@ const createDispatcher = () => {
...
     ) => {
-      if (type === 'title') {
-        titleQueue[titleQueue.indexOf(prevPayload as string)] = payload;
-        if (!isServerSide)
-          document.title = applyTitleTemplate(payload, titleTemplateQueue[0]);
-      } else if (type === 'titleTemplate') {
-        titleTemplateQueue[
-          titleTemplateQueue.indexOf(prevPayload as string)
-        ] = payload;
+      if (type === TITLE || type === TEMPLATE) {
```
Comment (JoviDeCroock) [inline src/dispatcher/index.ts:169]: CC @gustavpursche added some cases here to update head when we update title or template on index 0

File: src/dispatcher/index.ts [hunk 3]
```diff
@@ -100,35 +105,33 @@ const createDispatcher = () => {
...
-          );
-          break;
-
-        case 'meta':
-          metaQueue.splice(currentMetaIndex++, 0, payload as MetaPayload);
-          break;
-
-        default:
-          linkQueue.push(payload);
+      if (type === TITLE) {
```
Comment (JoviDeCroock) [inline src/dispatcher/index.ts:114]: Hope you don't mind me reverting this, it added a lot of size :/

File: src/hooks/useTitleTemplate.ts [hunk 4]
```diff
@@ -1,32 +1,3 @@
...
-    return () => {
-      hasMounted.current = false;
-      dispatcher._removeFromQueue(
-        'titleTemplate',
-        prevTitleTemplate.current as string
-      );
-    };
-  }, []);
-};
+export const useTitleTemplate = (template: string) => useTitle(template, true);
```
Comment (JoviDeCroock) [inline src/hooks/useTitleTemplate.ts:3]: How do you like this @gustavpursche

File: src/dispatcher/index.ts (hunk 3 above)
Comment (gustavpursche) [reply src/dispatcher/index.ts:114]: I don't. Thought it would be more readable :)

File: src/hooks/useTitleTemplate.ts [hunk 5]
```diff
@@ -1,32 +1,5 @@
...
-    );
-    return () => {
-      hasMounted.current = false;
-      dispatcher._removeFromQueue(
-        'titleTemplate',
-        prevTitleTemplate.current as string
-      );
-    };
-  }, []);
+  useTitle(template, true);
```
Comment (gustavpursche) [inline src/hooks/useTitleTemplate.ts:4]: :rocket: Beautiful.

**Comment by JoviDeCroock** [comment]
@gustavpursche oh no problem, I'll publish this tonight and I really couldn't be more grateful for taking your time to provide me with some feedback and this awesome feature-idea!

Have an awesome day and stay safe

**Pull request merged**